File: coupon/coupon-api/src/main/java/dev/be/coupon/api/coupon/application/dto/CouponCreateResult.java
package dev.be.coupon.api.coupon.application.dto;


import dev.be.coupon.domain.coupon.Coupon;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record CouponCreateResult(
        UUID id,
        String couponName,
        String couponType,
        String couponDiscountType,
        BigDecimal couponDiscountValue,
        int totalQuantity,
        String couponStatus,
        LocalDateTime expiredAt
) {
    public static CouponCreateResult from(final Coupon coupon) {
        return new CouponCreateResult(
                coupon.getId(),
                coupon.getCouponName().getName(),
                coupon.getCouponType().name(),
                coupon.getCouponDiscountType().name(),
                coupon.getCouponDiscountValue(),
                coupon.getTotalQuantity(),
                coupon.getCouponStatus().name(),
                coupon.getExpiredAt()
        );
    }
}

File: coupon/coupon-api/src/main/kotlin/dev/be/coupon/api/coupon/application/dto/CouponCreateResult.kt
package dev.be.coupon.api.coupon.application.dto

import dev.be.coupon.domain.coupon.Coupon
import java.time.LocalDateTime
import java.util.*

data class CouponCreateResult(
    val id: UUID,
    val name: String,
    val type: String,
    val totalQuantity: Int,
    val status: String,
    val validFrom: LocalDateTime,
    val validUntil: LocalDateTime
) {
    companion object {
        fun from(coupon: Coupon): CouponCreateResult {
            return CouponCreateResult(
                id = coupon.id,
                name = coupon.couponName.name,
                type = coupon.couponType.name,
                totalQuantity = coupon.totalQuantity,
                status = coupon.couponStatus.name,
                validFrom = coupon.validFrom,
                validUntil = coupon.validUntil
            )
        }
    }
}
